.pre-commit-config.yaml

```diff
@@ -13,3 +13,11 @@ repos:
     rev: "1.5.1"
     hooks:
       - id: poetry-check
+  - repo: https://github.com/commitizen-tools/commitizen
+    rev: v3.12.0
+    hooks:
+      - id: commitizen-branch
+        stages:
+          - push
+          - pre-merge-commit
+        args: ["--rev-range", "main..HEAD"]
```

Comment:
Reading this line and then thinking about commitzen made me wonder: are we leaving commitizen formatting of commits as a flexible option when it comes to making commits? For ex.:
- What would happen to PR's which don't abide this (would we disapprove solely based on the commit style)?
- Thinking about the end of the PR and getting ready to merge: how are squashes treated with this (would we lose the ability to leverage GitHub's built-in functionality here)?

Reply:
These are good points that I think merit further discussion. I'm going to remove this pre-commit hook for now and we can add it after 1.0 release.